// simple-react-native-color-picker/index.js
import React, { Component } from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Feather';

export default class ColorPicker extends Component {
    constructor(props){
        super(props);
        this.state = {selected: 0};

        this.getIndexSelected = this.getIndexSelected.bind(this);
        this.getColorSelected = this.getColorSelected.bind(this);
        this._select = this._select.bind(this);
    }

    getIndexSelected(){
        return this.state.selected;
    }
    getColorSelected(){
        return this.props.colors[this.state.selected];
    }

    _select(i){
        this.setState({...this.state, selected: i});
    }


    render() {
        if (!this.props.colors || !this.props.colors.length)
            return <View><Text>Ec!</Text></View>;


        let colors = this.props.colors.map((e, i) => {
            return (
                <View style={ styles.circleElementWrap } key={i}>
                    <TouchableOpacity onPress={()=>this._select(i)} style={[styles.circleElement, { backgroundColor: e }]}>
                        {
                            i == this.state.selected &&
                            <Icon name="check" size={40} color="rgba(0, 0, 0, 0.7)"></Icon>
                        }
                    </TouchableOpacity>
                </View>
            );
        });
        return (
            <View style={ styles.container }>
                { colors }
            </View>
        );
    }
}


const styles = StyleSheet.create({
    container: {
        alignSelf: 'stretch',
        justifyContent: 'center',
        flexDirection: 'row',
        flexWrap: 'wrap',
    },
    circleElement: {
        width: '100%',
        height: '100%',
        borderRadius: 50,
        justifyContent: 'center',
        alignItems: 'center'
    },
    circleElementWrap: {
        width: 70,
        height: 70,
        padding: 5
    }
});
